File: Homeworks/C#2/ArraysHomework/11.BinarySearch/BinarySearch.cs
/*
 Problem 11. Binary search

Write a program that finds the index of given element in a sorted array of integers by using the Binary search algorithm.
*/


using System;
using System.Collections.Generic;
using System.Linq;

class BinarySearch
{
    static void Main()
    {
        Console.Write("Please, enter number of elements: ");
        int n = int.Parse(Console.ReadLine());
        int[] binarrySearchedArray = new int[n];

        for (int i = 0; i < binarrySearchedArray.Length; i++)
        {
            Console.Write("Enter element[{0}]: ", i);
            binarrySearchedArray[i] = int.Parse(Console.ReadLine());
        }

        Console.Write("Please enter searched number: ");
        int searchedNumber = int.Parse(Console.ReadLine());

        int start = 0;
        int end = binarrySearchedArray.Length-1;

        while ((end - start) > 1)
        {
            int index = start + (end - start) / 2;

            if (binarrySearchedArray[index] == searchedNumber) 
            { 
                Console.WriteLine("{0} is at position {1}", searchedNumber, index); 
                return; 
            }
            else if (binarrySearchedArray[index] > searchedNumber)
            {
                end = index;
            }
            else if (binarrySearchedArray[index] < searchedNumber)
            {
                start = index;
            }
        }
        Console.WriteLine("Not found!");

    }
}

File: Homeworks/JavaScriptFundamentals/05.Arrays/07.BinarySearch/BinarySearch.js
/*
 Write a program that finds the index of given element in a sorted array
 of integers by using the Binary search algorithm.
 */

var jsConsole,
    array = [1, 6, 12, 15, 19, 21, 33, 45, 54, 60, 66, 76, 80, 96, 100, 104, 115, 123],
    button = document.getElementById('check'),
    searchNumber,
    start = 0,
    end = array.length,
    index;

button.onclick = getIndex;

jsConsole.writeLine('Test array is: [' + array + ']');
console.log('Test array is: [' + array + ']');

function getIndex() {
    searchNumber = jsConsole.readInteger('#search-number');

    while ((end - start) >= 1) {

        index = start + (((end - start) / 2) | 0);

        if (array[index] === searchNumber) {
            jsConsole.writeLine(searchNumber + ' is at position ' + index);
            console.log(searchNumber + ' is at position ' + index);
            return;
        }
        else if (array[index] > searchNumber) {
            end = index;
        }
        else if (array[index] < searchNumber) {
            start = index;
        }
    }

    jsConsole.writeLine("Number not found!");
    console.log("Number not found!");
}
